lang/c++/api/Serializer.hh

```diff
@@ -71,15 +68,15 @@ class Serializer : private boost::noncopyable
     }
 
     void writeBytes(const void *val, size_t size) {
-        writer_.writeBytes(val);
+        writer_.writeBytes(val, size);
```

Comment:
this seems like it should be tracked in a separate JIRA, it looks like an actual bug fix?  Is there an associated unit test?

Reply:
This is indeed a bug. The only caller for `writeBytes` is `serialize` in `AvroSerializer.hh`. `serialize` is not used by any of the code here, but could be used by some of generated code. Without this fix, the code can crash because there is no guarantee that the string is null-terminated.